roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -93,22 +89,14 @@ def run(self):
 
         return {}
 
-    @staticmethod
-    def free_bytes(path, ansible_mounts):
+    def free_bytes(self, path):
         """Return the size available in path based on ansible_mounts."""
-        mount_point = path
-        # arbitry value to prevent an infinite loop, in the unlike case that '/'
-        # is not in ansible_mounts.
-        max_depth = 32
-        while mount_point not in ansible_mounts and max_depth > 0:
-            mount_point = os.path.dirname(mount_point)
-            max_depth -= 1
-
+        mount = self.find_ansible_mount(path)
         try:
-            free_bytes = ansible_mounts[mount_point]['size_available']
+            return mount['size_available']
         except KeyError:
-            known_mounts = ', '.join('"{}"'.format(mount) for mount in sorted(ansible_mounts)) or 'none'
-            msg = 'Unable to determine disk availability for "{}". Known mount points: {}.'
-            raise OpenShiftCheckException(msg.format(path, known_mounts))
-
-        return free_bytes
+            raise OpenShiftCheckException(
+                'Unable to retrieve disk availability for "{}" due to a bug.\n'
```

Comment:
Hmm, as a user, I do not like a software telling me about it's own bugs, specially without any clue as to what the bug is and what can I do about it...
Suggestion: s/due to a bug//, s/Ansible// (do not blame the tool :P)

'Unable to compute disk availability for "{}". Missing key "size_available" ... got "{}".... You can retry running the check, or inspect the output of `ansible -m setup HOSTNAME` and make sure Ansible can report... etc.'

The user should understand from the message:

- What happened at a high level: "the disk availability check failed because it was unable to compute disk availability"
- Some indication of why it failed: "we have information about total_size, etc etc for the mount path /foo/bar, but we don't have 'size_available'" -- and we should hope this actually never happens.
- What can the user do next -- hmm, in this case maybe retry, if not there is the ansible -m setup... or report a bug...